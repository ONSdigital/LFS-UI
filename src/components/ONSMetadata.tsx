import React, {Component} from "react";

interface Props {
    List : any
    RightList? : any
}

export class ONSMetadata extends Component < Props, {} > {
    constructor(props : Props) {super(props);}
    
    render () {
        return(
            <div>
                <dl className="metadata metadata__list grid grid--gutterless u-cf u-mb-l" title="This is an example of the metadata component" aria-label="This is an example of the metadata component">
                    {this.props.List.map((item:any) => (<div>
                                                            <dt className="metadata__term grid__col col-2@m">{item.L}:</dt>  
                                                            <dd className="metadata__value grid__col col-10@m">{item.R}</dd>
                                                        </div>)
                    )}
                </dl>
            </div>
        )
    }
}