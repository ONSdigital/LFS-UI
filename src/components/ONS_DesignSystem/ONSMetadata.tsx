import React, {Component, Fragment} from "react";

interface Props {
    List : any
    LSpacing?: string
    RSpacing?: string
}


export class ONSMetadata extends Component < Props, {} > {
   
    render () {         
        let Lspacing = this.props.LSpacing;
        let Rspacing = this.props.RSpacing;

        if(this.props.LSpacing === undefined) Lspacing = "2";
        if(this.props.RSpacing === undefined) Rspacing = "10";
        
        if (this.props.List !== null) {
            return(
                <div>
                    <dl className="metadata metadata__list grid grid--gutterless u-cf u-mb-l" title="This is an example of the metadata component" aria-label="This is an example of the metadata component">
                        {this.props.List.map((item:any) => (
                            <Fragment key ={item.L}>
                                <dt className={"metadata__term grid__col col-" + Lspacing + "@m"}>{item.L}:</dt>
                                <dd className={"metadata__value grid__col col-" + Rspacing + "@m"}>{item.R}</dd>
                            </Fragment>)
                        )}
                    </dl>
                </div>
            )
        }

    }
}
