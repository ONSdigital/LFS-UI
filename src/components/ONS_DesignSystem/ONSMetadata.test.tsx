import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import {ONSMetadata} from "./ONSMetadata";
import Adapter from 'enzyme-adapter-react-16';

describe("ONS MetaData Test", () => {
    Enzyme.configure({ adapter: new Adapter() });
    
    const list = [{L: "Batch Type", R: "Monthly"},
                {L: "Year", R: "2015"}]

    const Props = {
        List: list,
        LSpacing: "10",
        RSpacing: "10"
    }

    const nullProps = {
        List: null
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSMetadata
                List={props.List}
                LSpacing={props.LSpacing}
                RSpacing={props.RSpacing}>
            </ONSMetadata>
            
        )
    }

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should render correctly", () => expect(wrapper(shallow, nullProps).exists()).toEqual(true));

    it('displays the correct left and right margins', () => {
        expect(wrapper(mount, Props).getElement().props.LSpacing).toEqual(Props.LSpacing)
        expect(wrapper(mount, Props).getElement().props.RSpacing).toEqual(Props.RSpacing)
    })

    it('displays the correct left and right margins', () => {
        expect(wrapper(mount, Props).at(0).find("dt")).toHaveLength(2)
    })



})