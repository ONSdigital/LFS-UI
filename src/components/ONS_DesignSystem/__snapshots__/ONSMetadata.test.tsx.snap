// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS MetaData Test matches Snapshot 1`] = `ShallowWrapper {}`;
